jobs/pacman-awsrules/src/test/java/com/tmobile/cloud/gcprules/cloudfunctions/HttpTriggersTest.java

```diff
@@ -46,40 +48,74 @@ public void setUp() {
+    private JsonArray getHitsJsonArrayForEnableHttpsRule() {
         Gson gson = new Gson();
         JsonObject jsonObject = new JsonObject();
+        //Changes made as per CQ/Mapper Generated data
         jsonObject.add("_source", gson.fromJson(
-                " {\n" +
-                        "          \"discoveryDate\": \"2022-12-08 17:00:00+0530\",\n" +
-                        "          \"_cloudType\": \"GCP\",\n" +
-                        "          \"region\": \"us-central1\",\n" +
-                        "          \"id\": \"projects/central-run-349616/locations/us-central1/functions/function-2\",\n" +
-                        "          \"projectName\": \"Paladin cloud\",\n" +
-                        "          \"projectId\": \"central-run-349616\",\n" +
-                        "          \"functionName\": \"projects/central-run-349616/locations/us-central1/functions/function-2\",\n" +
-                        "          \"ingressSetting\": \"ALLOW_ALL\",\n" +
-                        "          \"vpcConnector\": \"projects/central-run-349616/locations/us-central1/connectors/CONNECTOR_NAME\",\n" +
-                        "          \"httpTrigger\": \"1\",\n" +
-                        "          \"discoverydate\": \"2022-12-08 11:48:00+0000\"\n" +
-                        "        }\n" ,
+                "{" +
+                        "    \"ingressSetting\": \"ALLOW_ALL\"," +
+                        "    \"httpTrigger\": 1," +
+                        "    \"functionName\": \"projects/central-run-349616/locations/us-central1/functions/function-2\"," +
+                        "    \"discoverydate\": \"2023-10-13 21:27:00+0530\"," +
+                        "    \"vpcConnector\": \"\"," +
+                        "    \"_cloudType\": \"gcp\"," +
+                        "    \"id\": \"projects/central-run-349616/locations/us-central1/functions/function-2\"," +
+                        "    \"region\": \"us-central1\"," +
+                        "    \"projectId\": \"central-run-349616\"," +
+                        "    \"tags\": {" +
+                        "      \"deployment-tool\": \"console-cloud\"" +
+                        "    }" +
+                        "}",
                 JsonElement.class));
         JsonArray array = new JsonArray();
         array.add(jsonObject);
         return array;
     }
 
+    /**
+     * Mock the ES source data for asset type gcp_cloudfunction
+     *
+     * @return jsonObject contains mocked data(i.e Shipper ship mapped generated data, which be saved at elastic search under index "gcp_cloudfunction"
+     */
+    private JsonObject getJsonObjectForEnableHttpsRule() {
+        Gson gson = new Gson();
+        String mapperResultJsonForGcpFunction = "{" +
+                "    \"ingressSetting\": \"ALLOW_ALL\"," +
+                "    \"httpTrigger\": 1," +
+                "    \"functionName\": \"projects/central-run-349616/locations/us-central1/functions/function-2\"," +
```

Comment:
Can we generalize this? We should not be using the account name as it can change.

Reply:
This is only for UTs if I see it correctly.

Reply:
@plyubich @mkinchin-paladin This is just UT Test data for testing Mapper results for only cloud function because this policy is applicable to gcp-cloud function asset type. Thats why there is seperate test case exists for each policy for each asset type. So there is no need to generalise. Here we just want to make sure mapper has all the required attributes to run policy or not since there is no way to compare with legacy collector